ocaml/alerts/certificate/certificate_check.ml

```diff
@@ -1,8 +1,17 @@
 module XenAPI = Client.Client
 module Date = Xapi_stdext_date.Date
 
+type entity = Host of API.ref_host | Pool of API.ref_pool
+
+type cert = {entity: entity; expiry: API.datetime}
```

Comment:
There are three different messages, depending on the type of certificate. The current code only differentiates between two types. The certificates for internal communication use different messages because the way to resolve its expiry is different from the normal one.

> `HOST_INTERNAL_CERTIFICATE_EXPIRING_30`, `HOST_INTERNAL_CERTIFICATE_EXPIRING_14`, `HOST_INTERNAL_CERTIFICATE_EXPIRING_07`, `HOST_INTERNAL_CERTIFICATE_EXPIRED`: Similar to host certificates, the newly-introduced hosts' internal server certificates are monitored for expiry dates and alerts are generated about them. The body for this type of message is:
> `<body><message>The TLS server certificate for internal communications {is expiring soon|has expired}.</message><date>20210302T02:00:01Z</date></body>`

Reply:
The existing code is tedious by not being parameterized over type of certificate and expiration and we are increasing this problem. The internal certificate is always crated by us and I think a simpler handling can be justified.

Reply:
I've tried to make the handling simpler, the big issue here is the lack of changing the certificate provided by stunnel while it's running. This means it needs to be restarted, this means breaking active connections, with the risks of interrupting operations.

I would rather make users be aware of this fact rather than trying to automate it and risk corrupting data.